### DISCUSSION TITLE
Is it possible to store the targets cache in webhdfs?

### DISCUSSION THREAD

**Discussion by matthiasgomolka** [ropensci/targets#735] [Help] [answered]
I saw that it's possible to store data on Amazon S3 storage. This made me wonder if is possible to use a targets cache in a webhdfs directory. 

Since I haven't read anything about this before, I assume this is not (yet) possible. Would it be difficult to implement such a functionality? What would be needed to do this?

**Comment by wlandau** [comment, marked as answer]
Sorry, I do not have plans for webhdfs. `targets` only supports the most common types of storage because each storage format needs to be hard-coded. Too many types of storage would create too many exceptions and too many dependencies.

**Reply by matthiasgomolka** [reply]
I understand. In theory, would it be possible to put all that's necessary into another package which would be part of the targetopia? Or is this functionality buried too deep so that it would have to be implemented in targets itself?

**Reply by wlandau** [reply]
It's too internal because the way `targets` represents metadata depends on the quirks of the specific cloud platform, and that's a level of detail that I do not think should be exposed.

**Reply by matthiasgomolka** [reply]
Ok, thanks for clarifying!
